doc/howtos/building-images.rst

```diff
@@ -78,6 +81,51 @@ the existing user accounts (including root) has a password set, so
 logging into the running system is impossible. Before building an image,
 you must choose a way of interacting with the system after it has booted.
 
+.. NOTE: this section introduces the difference between development and production
+   images first because it is a choice that must be made before building. Choosing
+   architecture, image format and image content are more important than optional
+   build tweaks (sstate, removal of old images), so those come last.
+
+
+Development Images
+------------------
+
+All images provided by the Ostro Project are intended for
+developers and not directly for production use. To avoid having developers
+accidentally build images for real products that have development
+features enabled, you must make explicit changes in ``local.conf`` to
+enable them.
+
+Developers building their own images for personal use can follow these
+instructions to replicate the configuration of the published Ostro OS images. All necessary
+private keys are provided in the ``ostro-os`` repository.
+
+To do this, before building,  edit the :file:`conf/local.conf` configuration file,
+find the line
+with ``# require conf/distro/include/ostro-os-development.inc`` and
+uncomment it. This will also add some recommended software to the ``ostro-image-noswupd``
+reference image, see below for details.
+
+
+Production Images
+-----------------
+
+When building production images, first follow the instructions
+provided in :file:`meta-intel-iot-security/meta-integrity/README.md` for creating your own
```

Comment:
Yes, this should be included in the documentation set.

README files are OK for some explanatory information about the component, but we should not have users wandering around components looking for them.  We (Geoffroy/David) should take a look around the other components and see if there are substantial gems of information scattered around in README files we can incorporate into the documentation set.


-- david

From: Gustavo Sverzut Barbieri [mailto:notifications@github.com]
Sent: Tuesday, April 12, 2016 7:27 AM
To: ostroproject/meta-ostro <meta-ostro@noreply.github.com>
Cc: Kinder, David B <david.b.kinder@intel.com>
Subject: Re: [ostroproject/meta-ostro] simplify image content and bundles (#104)


In doc/howtos/building-images.rst<https://github.com/ostroproject/meta-ostro/pull/104#discussion_r59385123>:

> +Developers building their own images for personal use can follow these

> +instructions to replicate the configuration of the published Ostro OS images. All necessary

> +private keys are provided in the ``ostro-os`` repository.

> +

> +To do this, before building,  edit the :file:`conf/local.conf` configuration file,

> +find the line

> +with ``# require conf/distro/include/ostro-os-development.inc`` and

> +uncomment it. This will also add some recommended software to the ``ostro-image-noswupd``

> +reference image, see below for details.

> +

> +

> +Production Images

> +-----------------

> +

> +When building production images, first follow the instructions

> +provided in :file:`meta-intel-iot-security/meta-integrity/README.md` for creating your own

@dbkinder<https://github.com/dbkinder> and @gvancuts<https://github.com/gvancuts> should we convert that doc to .rst and then integrate it in our website autogen documentation?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/ostroproject/meta-ostro/pull/104/files/706737ae1b53d4576a0c68c3f4ebcb961063b620#r59385123>